### PR TITLE
Include in response only CSS variations that are needed

### PR DESCRIPTION
**Pull request by cperryk**
Ensures CSS variations that do not appear on the requested page do not appear in the response HTML.

### CODE REVIEW

File: lib/media.js [hunk 1]
```diff
@@ -160,91 +159,86 @@ function injectTags(fileArray, site, tag) {
...
-      });
-    }
-  }
-
-  return _.map(_.filter(cssFilePaths, files.fileExists), pathJoin(assetHost, assetPath, assetDir));
+  return state._components
+    .reduce((prev, name) => {
+      prev.push(
+        path.join(assetDir, 'js', `${name}.js`),
+        path.join(assetDir, 'js', `${name}.${slug}.js`),
```
Comment (amycheng) [inline lib/media.js]: don't think we have site-specific component js?

File: lib/media.js [hunk 2]
```diff
@@ -160,91 +159,86 @@ function injectTags(fileArray, site, tag) {
...
+    assetHost = site && site.assetHost || '',
+    siteStyleguide = site && site.styleguide;
+  let cssFilePaths;
+
+  // legacy behavior: load <component>.css and <component>.<site>.css
+  if (!site.styleguide) {
+    const siteSlug = site && site.slug;
+
+    cssFilePaths = state._components
+      .reduce((acc, curr) => {
```
Comment (amycheng) [inline lib/media.js]: what is the var `acc` referring to?

File: lib/media.js (hunk 1 above)
Comment (yuliyv) [reply lib/media.js]: This looks like it was pulled from the previous version that also got three js files for each component https://github.com/clay/amphora-html/blob/v3.3.0/lib/media.js#L192

I can't necessarily speak to why it was setup that way originally, but I suspect it was included here for compatibility reasons

File: lib/media.js (hunk 2 above)
Comment (yuliyv) [reply lib/media.js]: Pretty sure it's the local var for the accumulation of the reduction, could probably change this to `pathsToRetrieve` or something if that's more clear

**Pull request merged**